client/plots/disco_new/mapper/CnvArcsMapper.ts

```diff
@@ -0,0 +1,190 @@
+
+        this.onePxArcAngle = 1 / (settings.rings.cnvInnerRadius)
+
+        const gain = new CnvLegend("Max", (cnvMaxValue > 0)?CnvType.Gain: CnvType.Loss, this.getColor(cnvMaxValue), cnvMaxValue)
+        const loss = new CnvLegend("Min", (cnvMinValue > 0)?CnvType.Gain: CnvType.Loss, this.getColor(cnvMinValue), cnvMinValue)
+        const cap = new CnvLegend("Capping", CnvType.Loss, this.getColor((cnvMinValue > 0)? cnvMinValue : cnvMaxValue), this.settings.cnv.capping)
+
+        this.cnvClassMap.set(CnvType.Gain, gain)
+        this.cnvClassMap.set(CnvType.Loss, loss)
+        this.cnvClassMap.set(CnvType.Cap, cap)
+
+        this.maxAbsValue = Math.max(Math.abs(this.capMaxValue(cnvMinValue)), Math.abs(this.capMaxValue(cnvMaxValue)))
+    }
+
+    map(arcData: Array<Data>): Array<CnvArc> {
+        const arcs: Array<CnvArc> = []
+
+        arcData.forEach(data => {
+            let startAngle = this.calculateStartAngle(data)
+            let endAngle = this.calculateEndAngle(data)
+
+            if (endAngle - startAngle < this.onePxArcAngle) {
+                const restAngle = this.onePxArcAngle - (endAngle - startAngle)
+                startAngle = startAngle - restAngle / 2
+                endAngle = startAngle + restAngle / 2
+            }
+
+            const innerRadius = this.calculateInnerRadius(data)
+
+            const outerRadius = this.calculateOuterRadius(data)
+
+            const color = this.getColor(data.value)
+
+            const arc = new CnvArc(startAngle,
+                endAngle,
+                innerRadius,
+                outerRadius,
+                color,
+                data.gene,
+                data.chr,
+                data.start,
+                data.stop,
+                data.value,
+                this.cnvUnit
+            )
+
+            arcs.push(arc)
+        })
+
+        return arcs
+    }
+
+    calculateStartAngle(data: Data) {
+        const index = this.reference.chromosomesOrder.indexOf(data.chr)
+        const chromosome = this.reference.chromosomes[index]
+        return chromosome.startAngle + ((chromosome.endAngle - chromosome.startAngle) * (Number(data.start) / chromosome.size));
+    }
+
+    private calculateEndAngle(data: Data) {
+        const index = this.reference.chromosomesOrder.indexOf(data.chr)
+        const chromosome = this.reference.chromosomes[index]
+        return chromosome.startAngle + ((chromosome.endAngle - chromosome.startAngle) * (Number(data.stop) / chromosome.size));
+    }
+
+    getColor(value) {
+
+        const lossColor = this.settings.cnv.lossColor
+        const ampColor = this.settings.cnv.ampColor
+        const cappedLossColor = this.settings.cnv.cappedLossColor
+        const cappedAmpColor = this.settings.cnv.cappedAmpColor
+
+
+        if (value < this.lossCapped) {
+            return cappedLossColor
+        } else if (value <= 0) {
+            return lossColor
+        } else if (value <= this.gainCapped) {
```

Comment:
@aacic The condition here seems incorrect: is this supposed to be `else if (value >= this.gainCapped)`? Otherwise, if it's really less than, then it would seem random that a value could be matched against `< this.lossCapped` first. I could be wrong, without having read the input value as being either negative or positive, but it seems that the mutation class should be considered here.